common/src/main/java/com/wynntils/services/mapdata/MapDataService.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.services.mapdata;
+
+import com.wynntils.core.components.Service;
+import com.wynntils.services.mapdata.attributes.FullCategoryAttributes;
+import com.wynntils.services.mapdata.attributes.FullFeatureAttributes;
+import com.wynntils.services.mapdata.attributes.type.MapAttributes;
+import com.wynntils.services.mapdata.attributes.type.MapIcon;
+import com.wynntils.services.mapdata.providers.MapDataProvider;
+import com.wynntils.services.mapdata.type.MapCategory;
+import com.wynntils.services.mapdata.type.MapFeature;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+public class MapDataService extends Service {
+    private final MapDataProviders providers = new MapDataProviders();
+
+    public MapDataService() {
+        super(List.of());
+    }
+
+    public Stream<MapFeature> getFeatures() {
+        return providers.getProviders().flatMap(MapDataProvider::getFeatures);
+    }
+
+    public MapAttributes getFullFeatureAttributes(MapFeature feature) {
+        return new FullFeatureAttributes(feature);
+    }
+
+    public String getCategoryName(String categoryId) {
+        return getCategoryDefinitions(categoryId)
+                .map(MapCategory::getName)
+                .filter(Objects::nonNull)
+                .findFirst()
+                .orElse("NAMELESS CATEGORY");
```

Comment:
Fixed. I think this could need some additional polishing, how to best handle this extreme case (I'd almost like to call it an error, if we have categories without name).

Reply:
Yes, definitely a weird case..